File: multi-threaded-santa/python/src/multi_threaded_santa/elf_pool.py
class ElfPool:
    """Manages a fixed-size pool of elf workers.

    In Python, workers are threading.Thread instances.
    """

    def __init__(self, elf_count: int) -> None:
        if elf_count <= 0:
            raise ValueError("Elf pool must have at least one elf.")
        self.elf_count = elf_count

File: multi-threaded-santa/typescript/src/ElfPool.ts
/**
 * Manages a fixed-size pool of elf workers.
 * In Node.js, "workers" are concurrent async tasks, not OS threads.
 */
export class ElfPool {
  readonly elfCount: number;

  constructor(elfCount: number) {
    if (elfCount <= 0) {
      throw new Error('Elf pool must have at least one elf.');
    }
    this.elfCount = elfCount;
  }
}
